lib/view_component/compiler.rb

```diff
@@ -276,12 +281,17 @@ def normalized_variant_name(variant)
       variant.to_s.gsub("-", "__").gsub(".", "___")
     end
 
+    def safe_class_name
+      @safe_class_name ||= component_class.name.gsub("::", "_").underscore
```

Comment:
Will this result in conflicts between namespaces?

```
>> "Foo::Bar".gsub("::", "_").underscore
=> "foo_bar"
>> "FooBar".gsub("::", "_").underscore
=> "foo_bar"
```

```suggestion
      @safe_class_name ||= component_class.name.underscore.gsub("/", "__")
```

Thoughts on using ☝️ to avoid the conflict issues?

Reply:
Oh wow I definitely did not think of that! Good catch!